pkg/apis/stable/v1alpha1/allocationpolicy.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// AllocationPolicySpec defines the desired state of AllocationPolicy
+type AllocationPolicySpec struct {
+	// +kubebuilder:validation:Minimum=0
+	Priority int `json:"priority"`
+	// +kubebuilder:validation:Minimum=0
+	Weight         int                   `json:"weight"`
+	ConnectionInfo ClusterConnectionInfo `json:"connectionInfo,omitempty"`
+}
+
+// ClusterConnectionInfo defines the connection information for a cluster
+type ClusterConnectionInfo struct {
+	ClusterName             string        `json:"clusterName"`
+	APIServerEndpoint       string        `json:"apiServerEndpoint"`
+	Namespace               string        `json:"namespace"`
```

Comment:
I believe we can get rid of Namespace and AllocationServiceSecret if we just stored those as a "Secret". Here we can simply store the name of the secret.

If you look at what's stored in K8s service account token, it has ca.crt, namespace and secret already.

```
$ kubectl get secret -oyaml agones-sdk-token-n29lg
apiVersion: v1
data:
  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURDekNDQWZPZ0F3SUJBZ0lRUEdGRUN1a1ZrbThzenM4SnRMUUFOekFOQmdrcWhraUc5dzBCQVFzRkFEQXYKTVMwd0t3WURWUVFERXlSak9HTmxZMkk0WWkwMFl6QmlMVFEzWWprdFltVmlaaTFtTURneU5UTmhZVGxpTjJRdwpIaGNOTVRrd01qSXdNakl3TmpJNVdoY05NalF3TWpFNU1qTXdOakk1V2pBdk1TMHdLd1lEVlFRREV5UmpPR05sClkySTRZaTAwWXpCaUxUUTNZamt0WW1WaVppMW1NRGd5TlROaFlUbGlOMlF3Z2dFaU1BMEdDU3FHU0liM0RRRUIKQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUMyL3JKTm02d0hNRk9MMFFhQW5CUWw3b1B1YlR3elFqMTdNUWZmZHlhbwpDRk45RzFlRUhEZHRnSWlKY01SOGFPUCtRRm9Hb1lPL1ZmYXNOdm1kSE1ZRjRocnh1YkFOTWUvTmRsSWFLVno5Ci82d0NTZGlEaEI4T0p2Q0thaVRFZFdiaHhSWkFuR3dCMXZ5THZHL1hyYjNDMW1LME93TW82MWpRa1dXNE9BaGsKZEdNMHpJYnFlcTAyUWkzeExTY3JIQ01FSUdPVWVQV1RJQmUyVndjRy9KM29CRGwycjZOVURQaFFGZ0tkZTlocApNTW8vTDJncXFOeFhmVi9CZzFGc1NXU2xOYTR3VERKcFphNGVMN00rZkFwOWVRdzRoTXc2ODkzS3crQ253eG16Clk0Q2RWaCtyMnhkdGxsZmtRNWhjRVkydUswM09iR2hUQ1dnc1NWWUl4MU5iQWdNQkFBR2pJekFoTUE0R0ExVWQKRHdFQi93UUVBd0lDQkRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCQVFDcQpEeXluNFVuZWFuVUdFM294anBqRTNmN1BJOVFRNHI5UEZwdENDcUMvVkFpUVlPSTZuTUNLb3R5MmEzMVpDdnRBCjVUaEVPNlI2MitUVEovUDdINkppZHB4K3ZtMnoxMngvLzh5WS9FNVE3WGt6TUtROHYvNWFrem11NWl2Y25mRXQKS3U2NzF0ek5QVjVtc0MzYUtwNHh5NjlCNlA4cld1SUprQ2Y4NmtVanRHSnpYMXdNNGl1c2s2TndjR1hzL20zdQo1S0FEZm93VHVNNHNrUnk1ZEQ2VnJtL3BzcTJxUGk5V2xzbEliTHFUckhyR1VTODFPSUlTak5rTGZ4R0VETEpGCkVCM0hHUDBWQ0xtemdqQVBpR2N1T0c3bWZ5aVNBUGN1K3NWcE9QY2wrNUp1Qjh1STl4MnV2TzhjUEN6RzhWa2kKajVPbjVXQmFkd3BRZ3E3Qkp1L2IKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
  namespace: ZGVmYXVsdA==
  token: ZXlKaGJHY2lPaUpTVXpJMU5pSXNJbXRwWkNJNklpSjkuZXlKcGMzTWlPaUpyZFdKbGNtNWxkR1Z6TDNObGNuWnBZMlZoWTJOdmRXNTBJaXdpYTNWaVpYSnVaWFJsY3k1cGJ5OXpaWEoyYVdObFlXTmpiM1Z1ZEM5dVlXMWxjM0JoWTJVaU9pSmtaV1poZFd4MElpd2lhM1ZpWlhKdVpYUmxjeTVwYnk5elpYSjJhV05sWVdOamIzVnVkQzl6WldOeVpYUXVibUZ0WlNJNkltRm5iMjVsY3kxelpHc3RkRzlyWlc0dGJqSTViR2NpTENKcmRXSmxjbTVsZEdWekxtbHZMM05sY25acFkyVmhZMk52ZFc1MEwzTmxjblpwWTJVdFlXTmpiM1Z1ZEM1dVlXMWxJam9pWVdkdmJtVnpMWE5rYXlJc0ltdDFZbVZ5Ym1WMFpYTXVhVzh2YzJWeWRtbGpaV0ZqWTI5MWJuUXZjMlZ5ZG1salpTMWhZMk52ZFc1MExuVnBaQ0k2SWpGak1USmtNemxpTFRObU5qa3RNVEZsT1MwNU16WmpMVFF5TURFd1lUaGxNREJoT0NJc0luTjFZaUk2SW5ONWMzUmxiVHB6WlhKMmFXTmxZV05qYjNWdWREcGtaV1poZFd4ME9tRm5iMjVsY3kxelpHc2lmUS53eFkxVmNaa3BIWlFsRUZRN1FIMWd1NzdZR282eHplVjAzSE1OSlBUazlFbkFCMmRrWW5tQVY1bEpaZF9DZTFWeXFEMTBBSld1a0tOaVBDazl5RTZUUFJmaU9qZTBwWGZkNDJ0S3lzVDllcWZJaDdlOUxOR3I2dUtjU2xpam1jZVdDc1RmRUlSUUo3LUVjZXo0eEN5ajlHNDExQjhycWt3ei1JbW5VWXA2Vmdhc3VQNV9QYzRDR3N3TEYzMUdLdXFlNDJqSkNiV1RMV2xjbmxzZ3pmZ3BGQ3B6dV9GTDVFSm5JTm5GOFNWWXVrMkdjeU5rVjZiTnJzZV8zZGh1WXA2eHZOTDJScjdNR2FKR0RwdDNWSGVpUm9uOVFaM21jUTVzRk8tN3VUQnJkcHpWT0t5Wk5jWHJkQ2hUSEpIYjNtRm5zU1JzdHR0UWJvY2daaDhTZlp4YkE=
kind: Secret
metadata:
  annotations:
    kubernetes.io/service-account.name: agones-sdk
    kubernetes.io/service-account.uid: 1c12d39b-3f69-11e9-936c-42010a8e00a8
  creationTimestamp: "2019-03-05T17:07:08Z"
  name: agones-sdk-token-n29lg
  namespace: default
  resourceVersion: "9036083"
  selfLink: /api/v1/namespaces/default/secrets/agones-sdk-token-n29lg
  uid: 1c14e7e8-3f69-11e9-8947-42010a8e0089
type: kubernetes.io/service-account-token
```

Reply:
Great point. Thanks @jkowalski.
Done.